Release Sync to v0.9.0

Vincent056 opened the pull request.

mhils committed:
add `spoc install` command

pranitaT committed:
MULTIARCH-4654: Enabled the Security Profiles Operator for ppc64le, added fixes for seccomp and SELinux profiles, and verified functionality with logenricher.

Signed-off-by: Pranita Turrey <pranita.turrey@ibm.com>

ccojocar committed:
Harden the spo and bpf-recorder containers with custom apparmor profiles

Change-Id: Iccb89ec24d4f513acff9d7828dea6a4ab3c33ef1
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Fix typos

Change-Id: I8a4f7031c81fe1f47f1a0a55276b415bb6d59732
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Update seccomp and apaprmorprofile

Change-Id: Ie5183b2d1f0550ab463f4c3d0fd713d1de6ec39b
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Add the missing syscalls in the seccomp profile required to remove profiles

Change-Id: I913f1d20563311b5bb40b5f29293f1235a76a6d3
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Implement the apparmor_profile_audit_total metric

Change-Id: I53ac7c317d2a86d3cb086b189f707e492f3aff0b
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Ignore gocritic warning

Change-Id: I48d3f2c159da7e30e6745ba3c29a0565d5788818
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Add the profile name to apparmor_profile_error_total metric

Change-Id: I0706d004ac42d9df6c21a87b1c032c280ddae8a7
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Use a metric label name instead of reason inside of apparmor_profile_audit_total

Change-Id: I449d6c7a3b1a5e1c46b70bf57c84545d73124cbe
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Add a metric for apparmor profile denail

Add a metric which counts the total number of operations denined for a
apparmor profile.

Change-Id: I660500bf929cdd05e08a15a01681d5444739fc5b
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

mhils committed:
replace bpf_d_path_cursed with tetragon's implementation

mhils committed:
fix bpf_d_path compatibility

mhils committed:
remove `attr/apparmor/exec` from apparmor exclude list

mhils committed:
ci: improve logging on error

mhils committed:
skip all deleted files

mhils committed:
update crun profile

mhils committed:
fix logging

dcoppa committed:
bpf-recorder needs /sys/kernel/tracing/events/raw_syscalls/sys_enter/id

spod crashes when using the eBPF based recorder, because the container cannot access /sys/kernel/tracing/events/raw_syscalls/sys_enter/id:

I0122 08:11:14.334147    9234 bpfrecorder.go:517] "Excluding mount namespace" logger="bpf-recorder" mntns=4026531841
I0122 08:11:14.335822    9234 bpfrecorder.go:534] "BPF module successfully loaded." logger="bpf-recorder"
I0122 08:11:14.335896    9234 bpfrecorder.go:218] "Doing BPF start/stop self-test..." logger="bpf-recorder"
I0122 08:11:14.335967    9234 bpfrecorder.go:541] "Start BPF recording: Attaching all programs..." logger="bpf-recorder"
libbpf: failed to open '/sys/kernel/tracing/events/raw_syscalls/sys_enter/id': No such file or directory
libbpf: failed to determine tracepoint 'raw_syscalls/sys_enter' perf event ID: No such file or directory
libbpf: prog 'sys_enter': failed to create tracepoint 'raw_syscalls/sys_enter' perf event: No such file or directory
I0122 08:11:14.335948    9234 bpfrecorder.go:698] "Processing bpf events" logger="bpf-recorder"
E0122 08:11:14.336538    9234 main.go:240] "running security-profiles-operator" err="StartRecording self-test: attach base hooks: attach bpf program sys_enter: failed to attach program: no such file or directory" logger="setup"

mhils committed:
bpf recorder: attach programs on startup to win race with container

this is a follow-up to https://github.com/kubernetes-sigs/security-profiles-operator/pull/2675, which has proven not to be sufficient in practice.

this approach is worse performance-wise, but it's hard to say by how much. a follow-up commit will improve on this.

mhils committed:
bpf recorder: make `is_recording` more performant

mhils committed:
fixups

mhils committed:
make linter happy

mhils committed:
make linter happier

mhils committed:
prune runc container startup more agressively

ccojocar committed:
Fix typo in spo apparmor profile name

Change-Id: I68278e480d86f178b83184757588afc42c883951
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Update the security profiles for spo and bpf-recorder

Change-Id: I932a8a17580355346ad39712dbcef9d6ed53071c
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

dependabot[bot] committed:
build(deps): bump sigs.k8s.io/controller-runtime in the kubernetes group

Bumps the kubernetes group with 1 update: [sigs.k8s.io/controller-runtime](https://github.com/kubernetes-sigs/controller-runtime).


Updates `sigs.k8s.io/controller-runtime` from 0.20.0 to 0.20.1
- [Release notes](https://github.com/kubernetes-sigs/controller-runtime/releases)
- [Changelog](https://github.com/kubernetes-sigs/controller-runtime/blob/main/RELEASE.md)
- [Commits](https://github.com/kubernetes-sigs/controller-runtime/compare/v0.20.0...v0.20.1)

---
updated-dependencies:
- dependency-name: sigs.k8s.io/controller-runtime
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: kubernetes
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc from 1.69.4 to 1.70.0

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.69.4 to 1.70.0.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.69.4...v1.70.0)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mhils committed:
fix apparmor profile for spod

the template injection hack is not meant to be a permanent solution, but a temporary stopgap to make things work without reverts

dependabot[bot] committed:
build(deps): bump google.golang.org/protobuf from 1.36.3 to 1.36.4

Bumps google.golang.org/protobuf from 1.36.3 to 1.36.4.

---
updated-dependencies:
- dependency-name: google.golang.org/protobuf
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mhils committed:
make apparmor recording work with apparmor profile

saschagrunert committed:
Update golangci-lint and config

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

mhils committed:
make apparmor recording actually work

Co-authored-by: Cosmin Cojocar <ccojocar@google.com>

mhils committed:
apparmor recorder: generalize random digit sequences

mhils committed:
fix tests

dependabot[bot] committed:
build(deps): bump sigs.k8s.io/release-utils in the kubernetes group

Bumps the kubernetes group with 1 update: [sigs.k8s.io/release-utils](https://github.com/kubernetes-sigs/release-utils).


Updates `sigs.k8s.io/release-utils` from 0.9.0 to 0.10.0
- [Release notes](https://github.com/kubernetes-sigs/release-utils/releases)
- [Commits](https://github.com/kubernetes-sigs/release-utils/compare/v0.9.0...v0.10.0)

---
updated-dependencies:
- dependency-name: sigs.k8s.io/release-utils
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: kubernetes
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/containers/common from 0.61.1 to 0.62.0

Bumps [github.com/containers/common](https://github.com/containers/common) from 0.61.1 to 0.62.0.
- [Release notes](https://github.com/containers/common/releases)
- [Commits](https://github.com/containers/common/compare/v0.61.1...v0.62.0)

---
updated-dependencies:
- dependency-name: github.com/containers/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mhils committed:
apparmor recorder: improve generalization of random sequences

mhils committed:
bpf recorder: ensure resolved paths are null-terminated

dependabot[bot] committed:
build(deps): bump sigs.k8s.io/release-utils in the kubernetes group

Bumps the kubernetes group with 1 update: [sigs.k8s.io/release-utils](https://github.com/kubernetes-sigs/release-utils).


Updates `sigs.k8s.io/release-utils` from 0.10.0 to 0.11.0
- [Release notes](https://github.com/kubernetes-sigs/release-utils/releases)
- [Commits](https://github.com/kubernetes-sigs/release-utils/compare/v0.10.0...v0.11.0)

---
updated-dependencies:
- dependency-name: sigs.k8s.io/release-utils
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: kubernetes
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/cert-manager/cert-manager

Bumps [github.com/cert-manager/cert-manager](https://github.com/cert-manager/cert-manager) from 1.16.3 to 1.17.0.
- [Release notes](https://github.com/cert-manager/cert-manager/releases)
- [Changelog](https://github.com/cert-manager/cert-manager/blob/master/RELEASE.md)
- [Commits](https://github.com/cert-manager/cert-manager/compare/v1.16.3...v1.17.0)

---
updated-dependencies:
- dependency-name: github.com/cert-manager/cert-manager
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump ubi8/go-toolset from `9b7b1a4` to `bfe2159`

Bumps ubi8/go-toolset from `9b7b1a4` to `bfe2159`.

---
updated-dependencies:
- dependency-name: ubi8/go-toolset
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

mhils committed:
make linter happy

mhils committed:
fix profile install/uninstall path

mhils committed:
add `spoc remove` command

mhils committed:
bpf recorder: fix directory recording

With `/foo/**` in the profile, AppArmor still denies `mkdir('/foo')`. `**` does not match emptystr.

mhils committed:
extend spoc tests to check enforced profiles

mhils committed:
make the linter happy

mhils committed:
apparmor: do not deny mount if the container is doing raw io

mhils committed:
make the linter even happier

mhils committed:
force the linter into happiness

mhils committed:
try harder to make the linter happy

mhils committed:
try harderer to make the linter happy

mhils committed:
try hardererer to make the linter happy

dependabot[bot] committed:
build(deps): bump github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring

Bumps [github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring](https://github.com/prometheus-operator/prometheus-operator) from 0.79.2 to 0.80.0.
- [Release notes](https://github.com/prometheus-operator/prometheus-operator/releases)
- [Changelog](https://github.com/prometheus-operator/prometheus-operator/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus-operator/prometheus-operator/compare/v0.79.2...v0.80.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump golang.org/x/sync from 0.10.0 to 0.11.0

Bumps [golang.org/x/sync](https://github.com/golang/sync) from 0.10.0 to 0.11.0.
- [Commits](https://github.com/golang/sync/compare/v0.10.0...v0.11.0)

---
updated-dependencies:
- dependency-name: golang.org/x/sync
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump golang.org/x/mod from 0.22.0 to 0.23.0

Bumps [golang.org/x/mod](https://github.com/golang/mod) from 0.22.0 to 0.23.0.
- [Commits](https://github.com/golang/mod/compare/v0.22.0...v0.23.0)

---
updated-dependencies:
- dependency-name: golang.org/x/mod
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mhils committed:
improve test structure

mhils committed:
try to make the linter happy

mhils committed:
try harder to make the linter happy

mhils committed:
try harderer to make the linter happy

ccojocar committed:
Add fsck.ext4 to the list of allowed executables when mount is allowed

Change-Id: Ia43d16c6870b0713eba68e94bf5ed8d51e78b581
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

mhils committed:
try hardererer to make the linter happy

mhils committed:
bpf recorder: fix how we detect file types

ccojocar committed:
Add /sbin/fsck to allowed executables when mount is active, and generalize the /sys/devices path

Change-Id: I3f0ee4a91e6a59f6a868a1e581bcc231e85c7c7e
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

dependabot[bot] committed:
build(deps): bump google.golang.org/protobuf from 1.36.4 to 1.36.5

Bumps google.golang.org/protobuf from 1.36.4 to 1.36.5.

---
updated-dependencies:
- dependency-name: google.golang.org/protobuf
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

saschagrunert committed:
Update crun to 1.20

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

dependabot[bot] committed:
build(deps): bump ubi9/ubi-minimal from `b870979` to `66b9921`

Bumps ubi9/ubi-minimal from `b870979` to `66b9921`.

---
updated-dependencies:
- dependency-name: ubi9/ubi-minimal
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump golang.org/x/net from 0.34.0 to 0.35.0

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.34.0 to 0.35.0.
- [Commits](https://github.com/golang/net/compare/v0.34.0...v0.35.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump sigs.k8s.io/controller-tools in the kubernetes group

Bumps the kubernetes group with 1 update: [sigs.k8s.io/controller-tools](https://github.com/kubernetes-sigs/controller-tools).


Updates `sigs.k8s.io/controller-tools` from 0.17.1 to 0.17.2
- [Release notes](https://github.com/kubernetes-sigs/controller-tools/releases)
- [Changelog](https://github.com/kubernetes-sigs/controller-tools/blob/main/envtest-releases.yaml)
- [Commits](https://github.com/kubernetes-sigs/controller-tools/compare/v0.17.1...v0.17.2)

---
updated-dependencies:
- dependency-name: sigs.k8s.io/controller-tools
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: kubernetes
...

Signed-off-by: dependabot[bot] <support@github.com>

saschagrunert committed:
Update bundle

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

dependabot[bot] committed:
build(deps): bump the kubernetes group with 4 updates

Bumps the kubernetes group with 4 updates: [k8s.io/api](https://github.com/kubernetes/api), [k8s.io/apimachinery](https://github.com/kubernetes/apimachinery), [k8s.io/cli-runtime](https://github.com/kubernetes/cli-runtime) and [k8s.io/client-go](https://github.com/kubernetes/client-go).


Updates `k8s.io/api` from 0.32.1 to 0.32.2
- [Commits](https://github.com/kubernetes/api/compare/v0.32.1...v0.32.2)

Updates `k8s.io/apimachinery` from 0.32.1 to 0.32.2
- [Commits](https://github.com/kubernetes/apimachinery/compare/v0.32.1...v0.32.2)

Updates `k8s.io/cli-runtime` from 0.32.1 to 0.32.2
- [Commits](https://github.com/kubernetes/cli-runtime/compare/v0.32.1...v0.32.2)

Updates `k8s.io/client-go` from 0.32.1 to 0.32.2
- [Changelog](https://github.com/kubernetes/client-go/blob/master/CHANGELOG.md)
- [Commits](https://github.com/kubernetes/client-go/compare/v0.32.1...v0.32.2)

---
updated-dependencies:
- dependency-name: k8s.io/api
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: kubernetes
- dependency-name: k8s.io/apimachinery
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: kubernetes
- dependency-name: k8s.io/cli-runtime
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: kubernetes
- dependency-name: k8s.io/client-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: kubernetes
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/cert-manager/cert-manager

Bumps [github.com/cert-manager/cert-manager](https://github.com/cert-manager/cert-manager) from 1.17.0 to 1.17.1.
- [Release notes](https://github.com/cert-manager/cert-manager/releases)
- [Changelog](https://github.com/cert-manager/cert-manager/blob/master/RELEASE.md)
- [Commits](https://github.com/cert-manager/cert-manager/compare/v1.17.0...v1.17.1)

---
updated-dependencies:
- dependency-name: github.com/cert-manager/cert-manager
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump sigs.k8s.io/controller-runtime in the kubernetes group

Bumps the kubernetes group with 1 update: [sigs.k8s.io/controller-runtime](https://github.com/kubernetes-sigs/controller-runtime).


Updates `sigs.k8s.io/controller-runtime` from 0.20.1 to 0.20.2
- [Release notes](https://github.com/kubernetes-sigs/controller-runtime/releases)
- [Changelog](https://github.com/kubernetes-sigs/controller-runtime/blob/main/RELEASE.md)
- [Commits](https://github.com/kubernetes-sigs/controller-runtime/compare/v0.20.1...v0.20.2)

---
updated-dependencies:
- dependency-name: sigs.k8s.io/controller-runtime
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: kubernetes
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/opencontainers/runc from 1.2.4 to 1.2.5

Bumps [github.com/opencontainers/runc](https://github.com/opencontainers/runc) from 1.2.4 to 1.2.5.
- [Release notes](https://github.com/opencontainers/runc/releases)
- [Changelog](https://github.com/opencontainers/runc/blob/v1.2.5/CHANGELOG.md)
- [Commits](https://github.com/opencontainers/runc/compare/v1.2.4...v1.2.5)

---
updated-dependencies:
- dependency-name: github.com/opencontainers/runc
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump ubi9/ubi-minimal from `66b9921` to `14f14e0`

Bumps ubi9/ubi-minimal from `66b9921` to `14f14e0`.

---
updated-dependencies:
- dependency-name: ubi9/ubi-minimal
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/prometheus/client_golang

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.20.5 to 1.21.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/v1.21.0/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.20.5...v1.21.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

ccojocar committed:
Change the scope of security profiles CRDs to be cluster wide

Change-Id: I3aac182eb0afea485fd8e1488a0426265dbdd3f9
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

dependabot[bot] committed:
build(deps): bump github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring

Bumps [github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring](https://github.com/prometheus-operator/prometheus-operator) from 0.80.0 to 0.80.1.
- [Release notes](https://github.com/prometheus-operator/prometheus-operator/releases)
- [Changelog](https://github.com/prometheus-operator/prometheus-operator/blob/v0.80.1/CHANGELOG.md)
- [Commits](https://github.com/prometheus-operator/prometheus-operator/compare/v0.80.0...v0.80.1)

---
updated-dependencies:
- dependency-name: github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

ccojocar committed:
Remove the namespace from example profiles used by the e2e

Change-Id: I7fa534ff135e099cf045e59feb570c974fb4aa3e
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Update bundle

Change-Id: I4a8c8343c360ea8835effe94b90dded6da624053
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Update the documentation to reflect the cluster scope for security
profiles

Change-Id: Idd5bcf88cee3c53f7a7b3a22fd7d58750a6d2b9b
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Adapt the olm e2e tests to run with a cluster wide seccompprofile CR

Change-Id: I4725a64d12d51adc4d82d68d89a50e702cf57358
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Clean profiles namesapce from e2e tests

Change-Id: I620fbf8e02a72f94a1f32586b775ea9c48ed15b5
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Clean up the unsed namesapce from e2e functions

Change-Id: I64497103b12ff8773a46c90c95bc260dd9178e92
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
More cleanup

Change-Id: Ib755361811381bc12769a17926a8ece42564d4ab
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Fix logenricher e2e test

Change-Id: I7ecc5fbc6ff911912981ba9a4e9fb1889ee4270a
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Remove the namespace from e2e tests

Change-Id: I75c736be4c103543c963152a6be143c3f0418c4e
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Fix lint warning

Change-Id: Ic46ac8f56d0e28f1f1d477f1230cead7229d6d5c
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

dependabot[bot] committed:
build(deps): bump github.com/google/go-cmp from 0.6.0 to 0.7.0

Bumps [github.com/google/go-cmp](https://github.com/google/go-cmp) from 0.6.0 to 0.7.0.
- [Release notes](https://github.com/google/go-cmp/releases)
- [Commits](https://github.com/google/go-cmp/compare/v0.6.0...v0.7.0)

---
updated-dependencies:
- dependency-name: github.com/google/go-cmp
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

ccojocar committed:
Exit immediately when a command fails in e2e-olm tests

Change-Id: If72ac060fbe381c2e2a1f889fff1d5908286ec48
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Remove checks when using a custom namespace

Change-Id: I4ee9ea50a4c4e65fbfb82db4532274d5e65bcc87
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

ccojocar committed:
Disable the eBPF recorder in fedora e2e tests since seccomp is disabled

Change-Id: I45e2ae0529f54adab7b15899c8b5276776266964
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

dependabot[bot] committed:
build(deps): bump github.com/go-jose/go-jose/v4 from 4.0.4 to 4.0.5

Bumps [github.com/go-jose/go-jose/v4](https://github.com/go-jose/go-jose) from 4.0.4 to 4.0.5.
- [Release notes](https://github.com/go-jose/go-jose/releases)
- [Changelog](https://github.com/go-jose/go-jose/blob/main/CHANGELOG.md)
- [Commits](https://github.com/go-jose/go-jose/compare/v4.0.4...v4.0.5)

---
updated-dependencies:
- dependency-name: github.com/go-jose/go-jose/v4
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump ubi8/go-toolset from `bfe2159` to `8a9106c`

Bumps ubi8/go-toolset from `bfe2159` to `8a9106c`.

---
updated-dependencies:
- dependency-name: ubi8/go-toolset
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

ccojocar committed:
Remove the namespace from seccompprofile reference

Change-Id: Ie1b81affb79d9530da9b56afbdec523a78fe5695
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

saschagrunert committed:
Update go to v1.24

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

ccojocar committed:
Remove the namespace from security profiles path

Change-Id: I34c8a88d0af12a567b03c968fa1cdfe68ae68fc4
Signed-off-by: Cosmin Cojocar <ccojocar@google.com>

saschagrunert committed:
Bump to v0.9.0

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

saschagrunert reviewed (approved):
/retest

rhmdnd commented:
Review the 1.23 bump in https://github.com/openshift/release/pull/64599

Are we good to rekick the tests once that lands?

Vincent056 commented:
/retest

Vincent056 commented:
/retest

ngopalak-redhat committed:
Cherry Pick Fix RHEL 9 for openshift dev

Vincent056 commented:
Pulled in [MULTIARCH-4654: Enabled the Security Profiles Operator for ppc64le](https://github.com/openshift/security-profiles-operator/pull/60/commits/4c65c7733362d8a987604e7a50cf4bc6da51dea4)

rhmdnd reviewed (approved):
/lgtm

Vincent056 committed:
Migrate Tekton pipelines to new branching stretegy

    Switches multi-arch and bundle Tekton pipeline definitions to new files and conventions, aligning with rhel9-4-els/rhel-minimal as the build base

    Removes legacy pipeline config, adds new ones for Konflux branch, and updates rpm lockfile to match rhel9-4-els. Ensures consistent build and security scanning workflows across build and bundle images.

Vincent056 commented:
/retest

Vincent056 committed:
Bumps SPO version to 0.9.0 in build dockerfile configuration

Vincent056 committed:
Update konflux config to use -dev application

rhmdnd commented:
Looks like the e2e tests snowballed, resulting in a bunch of issues:

```
 time="2025-05-28T14:12:37Z" level=info msg="+ /cli/oc debug -q node/fake-node -- chroot /host /bin/bash -c test -f /var/lib/security-profiles-operator/delete-me.json"
Error from server (NotFound): nodes "fake-node" not found 
```

Vincent056 commented:
/retest

Vincent056 committed:
Update konflux tekton config for new konflux cluster

Vincent056 commented:
/retest

Vincent056 commented:
/retest

Vincent056 commented:
/retest

xiaojiey commented:
/lgtm

Vincent056 commented:
/retest

Vincent056 commented:
/retest

Vincent056 commented:
/retest

Vincent056 commented:
/retest

Vincent056 committed:
update konflux pipeline

Vincent056 commented:
/retest

rhmdnd commented:
Looks like the EC contract is failing `labels.required_labels` validations.

Vincent056 commented:
/retest

Vincent056 committed:
Add additional labels to the OpenShift Security Profiles Operator Dockerfile

This commit introduces new labels including release version, URL, vendor, distribution scope, and a description to the Dockerfile, enhancing metadata for the Security Profiles Operator. These are needed by EC test.

Vincent056 committed:
Refactor CSV image replacement logic

We need to replace all image reference in CSV, this commit adds that.

rhmdnd commented:
/retest

Vincent056 commented:
/retest
